boa_engine/src/builtins/mod.rs

```diff
@@ -276,6 +279,21 @@ impl Realm {
             intl::PluralRules::init(self);
         }
     }
+
+    /// Create a new Realm with the default global bindings.
+    pub fn create_with_default_global_bindings(
+        hooks: &dyn HostHooks,
+        root_shape: &RootShape,
+        context: &mut Context<'_>,
+    ) -> JsResult<Self> {
+        let realm = Self::create(hooks, root_shape);
+
+        let old_realm = context.enter_realm(realm);
+
+        set_default_global_bindings(context)?;
+
+        Ok(context.enter_realm(old_realm))
+    }
```

Comment:
Can this be moved to `Context` instead? It makes better sense if we associate the creation of a `Realm` with the `Context`, like `context.create_realm()?;` That would also allow us to make `Realm::create` crate-only, which avoids the pitfalls of using that API without a proper knowledge of the internals.

Reply:
Implemented in c5efb4118520a8d7fb392ab6dba532994073d806